File: dotnet_3/em/NCS/EmbeddedEvoMasterController.cs
using System;
using System.Collections.Generic;
using EvoMaster.Controller;
using EvoMaster.Controller.Api;
using EvoMaster.Controller.Problem;

namespace NCS
{
    public class EmbeddedEvoMasterController : EmbeddedSutController
    {
        private bool isSutRunning;
        private int sutPort;
        
        static void Main(string[] args)
        {
            var embeddedEvoMasterController = new EmbeddedEvoMasterController ();

            InstrumentedSutStarter instrumentedSutStarter = new InstrumentedSutStarter (embeddedEvoMasterController);

            System.Console.WriteLine ("Driver is starting...\n");

            instrumentedSutStarter.Start ();
        }

        public override void StopSut()
        {
            throw new NotImplementedException();
        }

        public override bool IsSutRunning()
        {
            throw new NotImplementedException();
        }

        public override string GetPackagePrefixesToCover()
        {
            throw new NotImplementedException();
        }

        public override List<AuthenticationDto> GetInfoForAuthentication()
        {
            throw new NotImplementedException();
        }

        public override string GetDatabaseDriverName()
        {
            throw new NotImplementedException();
        }

        public override IProblemInfo GetProblemInfo()
        {
            throw new NotImplementedException();
        }

        public override OutputFormat GetPreferredOutputFormat()
        {
            throw new NotImplementedException();
        }

        public override void ResetStateOfSut()
        {
            throw new NotImplementedException();
        }

        public override string StartSut()
        {
            throw new NotImplementedException();
        }
    }
}

File: jdk_8_maven/em/embedded/rest/genome-nexus/src/main/java/em/embedded/org/cbioportal/genome_nexus/EmbeddedEvoMasterController.java
package em.embedded.org.cbioportal.genome_nexus;

import com.mongodb.MongoClient;
import org.cbioportal.genome_nexus.GenomeNexusAnnotation;
import org.evomaster.client.java.controller.EmbeddedSutController;
import org.evomaster.client.java.controller.InstrumentedSutStarter;
import org.evomaster.client.java.controller.api.dto.auth.AuthenticationDto;
import org.evomaster.client.java.controller.api.dto.SutInfoDto;
import org.evomaster.client.java.sql.DbSpecification;
import org.evomaster.client.java.controller.problem.ProblemInfo;
import org.evomaster.client.java.controller.problem.RestProblem;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.testcontainers.containers.GenericContainer;

import java.util.Collections;
import java.util.List;
import java.util.Map;


/**
 * Class used to start/stop the SUT. This will be controller by the EvoMaster process
 */
public class EmbeddedEvoMasterController extends EmbeddedSutController {

    public static void main(String[] args) {

        int port = 40100;
        if (args.length > 0) {
            port = Integer.parseInt(args[0]);
        }

        EmbeddedEvoMasterController controller = new EmbeddedEvoMasterController(port);
        InstrumentedSutStarter starter = new InstrumentedSutStarter(controller);

        starter.start();
    }


    private ConfigurableApplicationContext ctx;

    private static final int MONGODB_PORT = 27017;

    private static final String MONGODB_VERSION = "3.6.2";

    private static final String MONGODB_DATABASE_NAME = "annotator";

    private static final GenericContainer mongodbContainer = new GenericContainer("mongo:" + MONGODB_VERSION)
            .withTmpFs(Collections.singletonMap("/data/db", "rw"))
            .withExposedPorts(MONGODB_PORT);


    private MongoClient mongoClient;


    public EmbeddedEvoMasterController() {
        this(0);
    }

    public EmbeddedEvoMasterController(int port) {
        setControllerPort(port);
    }


    @Override
    public String startSut() {

        mongodbContainer.start();

        try {
            mongoClient = new MongoClient(mongodbContainer.getContainerIpAddress(),
                    mongodbContainer.getMappedPort(MONGODB_PORT));

        } catch (Exception e) {
            System.out.println("ERROR: " + e.getMessage());
            throw new RuntimeException(e);
        }

        ctx = SpringApplication.run(GenomeNexusAnnotation.class,
                new String[]{"--server.port=0",
                        "--spring.data.mongodb.uri=mongodb://" + mongodbContainer.getContainerIpAddress() + ":" + mongodbContainer.getMappedPort(MONGODB_PORT) + "/" + MONGODB_DATABASE_NAME,
                        "--spring.cache.type=NONE"
                });

        return "http://localhost:" + getSutPort();
    }

    protected int getSutPort() {
        return (Integer) ((Map) ctx.getEnvironment()
                .getPropertySources().get("server.ports").getSource())
                .get("local.server.port");
    }


    @Override
    public boolean isSutRunning() {
        return ctx != null && ctx.isRunning();
    }

    @Override
    public void stopSut() {
        ctx.stop();
        ctx.close();

        mongoClient.close();
        mongodbContainer.stop();
    }

    @Override
    public String getPackagePrefixesToCover() {
        return "org.cbioportal.genome_nexus.";
    }

    @Override
    public void resetStateOfSUT() {
        mongoClient.getDatabase(MONGODB_DATABASE_NAME).drop();
    }


    @Override
    public List<DbSpecification> getDbSpecifications() {
        return null;
    }


    @Override
    public List<AuthenticationDto> getInfoForAuthentication() {
        return null;
    }


    @Override
    public ProblemInfo getProblemInfo() {
        return new RestProblem(
                "http://localhost:" + getSutPort() + "/v2/api-docs",
                null
        );
    }

    @Override
    public SutInfoDto.OutputFormat getPreferredOutputFormat() {
        return SutInfoDto.OutputFormat.JAVA_JUNIT_4;
    }

    @Override
    public Object getMongoConnection() {return mongoClient;}
}
